### PR TITLE
Empty dataset support

### PR DESCRIPTION
**Pull request by douglas1850**
*Issue #, if available:*

*Description of changes:* Changes to allow for datasets to be created at the time of translate/transform/training jobs being started. Also made a few small fixes to clean up things the linter was picking up like undefined values, deleting non-optional values, etc.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AlejandroRigau — DISMISSED** [review]

File: frontend/src/entities/dataset/dataset.service.ts [hunk 1]
```diff
@@ -185,28 +185,37 @@ export async function uploadFiles (
...
+    let unexpectedError = false;
+    createDataset(dataset).catch((error) => {
+        const expectedError = `Bad Request: Dataset ${dataset.name} already exists.`;
+        // Any error that the dataset already existing is unexpected
+        unexpectedError = expectedError !== error.response.data;
+    });
+    
+    return unexpectedError;
+};
+
```
Comment (KyleLilly) [inline frontend/src/entities/dataset/dataset.service.ts:218]: Why not just swallow expected errors and throw the unexpected ones? Is there a reason we're returning it and relying on the caller to know that/do something with it?

File: frontend/src/entities/jobs/hpo/create/hpo-job-create.tsx [hunk 2]
```diff
@@ -351,7 +354,29 @@ export function HPOJobCreate () {
...
+                    if (split[3] === 'private') {
+                        datasetType = DatasetType.PRIVATE;
+                    }
+                    const datasetName = split[6].replace('"', '');
+                    const newDataset = {
+                        name: datasetName,
+                        description: `Dataset created as part of the HPO job: ${state.form.HyperParameterTuningJobName}`,
+                        type: datasetType,
+                        scope: scope
+                    } as IDataset;
```
Comment (KyleLilly) [inline frontend/src/entities/jobs/hpo/create/hpo-job-create.tsx:370]: Cab we use `datasetFromS3Uri()` here?

File: frontend/src/entities/dataset/dataset.service.ts (hunk 1 above)
Comment (douglas1850) [reply frontend/src/entities/dataset/dataset.service.ts:218]: You're thinking I do the expected error check and if it's an unexpected error I just re-raise it, and then have the call to `createDatasetHandleAlreadyExists` be invoked in a try/catch right?

File: frontend/src/entities/jobs/hpo/create/hpo-job-create.tsx (hunk 2 above)
Comment (douglas1850) [reply frontend/src/entities/jobs/hpo/create/hpo-job-create.tsx:370]: Good call, wasn't aware of this function but it serves the same purpose.

**Review by KyleLilly — APPROVED** [review]

**Pull request merged**